crates/bevy_reflect/src/type_info.rs

```diff
@@ -146,6 +146,22 @@ impl TypeInfo {
     pub fn is<T: Any>(&self) -> bool {
         TypeId::of::<T>() == self.type_id()
     }
+
+    /// The docstring of the underlying type, if any.
+    #[cfg(feature = "documentation")]
+    pub fn doc(&self) -> Option<&str> {
+        match self {
+            Self::Struct(info) => info.doc(),
+            Self::TupleStruct(info) => info.doc(),
+            Self::Tuple(info) => info.doc(),
```

Comment:
Some of these are not actually useful to us (e.g. we'd almost never document a tuple), but I added them anyways to allow manual implementors to do their own thing if desired.

Can change if truly deemed unnecessary.